pkg/gitprovider/bitbucketserver.go

```diff
@@ -0,0 +1,402 @@
+			SourceRepoOwner: repoOwner,
+			SourceRepoName:  pr.FromRef.Repository.Name,
+		})
+	}
+
+	return response, nil
+}
+
+func (g *BitbucketServerGitProvider) GetUser() (*GitUser, error) {
+	client, err := g.getApiClient()
+	if err != nil {
+		return nil, err
+	}
+
+	// Since BitbucketServer or gfleury/go-bitbucket-v1 doesn't offer an endpoint to query the
+	// currently authenticated user, We instead query the '/rest/api/1.0/application-properties' endpoint
+	// which does not put load on the server and then extract the username from the response header.
+	// Refer this developer community comment: https://community.developer.atlassian.com/t/obtain-authorised-users-username-from-api/24422/2
+	res, err := client.DefaultApi.GetApplicationProperties()
+	if err != nil {
+		return nil, err
+	}
+
+	username := res.Header.Get("X-Ausername")
+	if username == "" {
+		return nil, fmt.Errorf("X-Ausername header is missing")
+	}
+
+	user, err := client.DefaultApi.GetUser(username)
+	if err != nil {
+		return nil, err
+	}
+
+	if user.Values == nil {
+		return nil, fmt.Errorf("User values are nil")
+	}
+	var userInfo bitbucketv1.User
+	err = mapstructure.Decode(user.Values, &userInfo)
+
+	if err != nil {
+		return nil, err
+	}
+
+	response := &GitUser{}
+	response.Id = fmt.Sprintf("%d", userInfo.ID)
+	response.Username = username
+	response.Name = userInfo.DisplayName
+
+	return response, nil
+}
+
+func (g *BitbucketServerGitProvider) GetLastCommitSha(staticContext *StaticGitContext) (string, error) {
+	client, err := g.getApiClient()
+	if err != nil {
+		return "", err
+	}
+
+	until := ""
+	if staticContext.Sha != nil {
+		until = *staticContext.Sha
+	}
+
+	commits, err := client.DefaultApi.GetCommits(staticContext.Id, staticContext.Name, map[string]interface{}{
+		"until": until,
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	if len(commits.Values) == 0 {
+		return "", fmt.Errorf("No commits found")
+	}
+
+	commitList, err := bitbucketv1.GetCommitsResponse(commits)
+	if err != nil {
+		return "", err
+	}
+
+	return commitList[0].ID, nil
+}
+
+func (g *BitbucketServerGitProvider) getPrContext(staticContext *StaticGitContext) (*StaticGitContext, error) {
+	if staticContext.PrNumber == nil {
+		return staticContext, nil
+	}
+
+	repo := *staticContext
+
+	client, err := g.getApiClient()
+	if err != nil {
+		return nil, err
+	}
+
+	pr, err := client.DefaultApi.GetPullRequest(staticContext.Id, staticContext.Name, int(*staticContext.PrNumber))
+	if err != nil {
+		return nil, err
+	}
+
+	prInfo, err := bitbucketv1.GetPullRequestResponse(pr)
+	if err != nil {
+		return nil, err
+	}
+
+	if prInfo.FromRef.Repository.Owner != nil {
+		ownerName := prInfo.FromRef.Repository.Owner.DisplayName
+		repo.Owner = ownerName
+		repo.Id = ownerName
+	}
+	repo.Name = prInfo.FromRef.Repository.Slug
+	repo.Branch = &prInfo.FromRef.DisplayID
+
+	return &repo, nil
+}
+
+func (g *BitbucketServerGitProvider) parseStaticGitContext(repoUrl string) (*StaticGitContext, error) {
```

Comment:
There seems to be an issue when parsing paths that have an anchor at the end. The anchor is produced by the Server to specify which file is being viewed.
![Screenshot 2024-06-26 at 09 36 33](https://github.com/daytonaio/daytona/assets/26512078/1472c0c7-8e58-4510-97f7-e61125e4036b)

Once I removed `#test`, the request went through.

Reply:
fixed, added a test case..

Reply:
@Tpuljak